Formula/optool.rb

```diff
@@ -0,0 +1,29 @@
+class Optool < Formula
+  desc "Command-Line tool for interacting with MachO binaries on macOS/iOS"
+  homepage "https://github.com/alexzielenski/optool"
+  # pull from git tag to get submodules
+  url "https://github.com/alexzielenski/optool.git",
+      tag:      "0.1",
+      revision: "a20c514baa2018adf53a0a408b9ee72cba107155"
+  license "BSD-2-Clause"
+  head "https://github.com/alexzielenski/optool.git"
+
+  depends_on xcode: :build
+
+  def install
+    xcodebuild "SDKROOT=",
+               "SYMROOT=build", "OBJROOT=build", "DSTROOT=build",
+               "-project", "optool.xcodeproj",
+               "-target", "optool",
+               "-configuration", "Release",
+               "OTHER_LDFLAGS='-ObjC'"
+    bin.install "build/Release/optool"
+  end
+
+  test do
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hi @SMillerDev,

thanks for your input regarding test sufficiency.

As a solution I extended the test section, adding comprehensive and encompassing cases.
Two primary application commands (`install`, `uninstall`) are executed.

Looking forward to your feedback, Cheers!
S